test/fork/integration.test.js

```diff
@@ -194,4 +194,52 @@ describe('Babylon integrations', function () {
     await increaseTime(ONE_DAY_IN_SECONDS * 30);
     await customStrategy.connect(keeper).finalizeStrategy(0, '', 0);
   });
+
+  it('can deploy a strategy with the Idle.Finance Custom integration', async () => {
```

Comment:
would be good to have two arrays with all the vaults. One for best yield and one for senior tranches and then have a loop for each and then run the tests for each vault.

bestYieldVaults.forEach((v) => {
  it('can enter into the Idle vault ${v}', () => {
   ...
  }); 
})

Reply:
This is done, however I'm still a bit confused about how Idle works now.

I added the array for the "best yield" vaults and they all seem to pass the tests.

However it looks like the "Perpetual Yield Tranches" is a different product completely: https://docs.idle.finance/developers/perpetual-yield-tranches/architecture

Seems like maybe this needs to be a separate Babylon integration, since it's a completely different interface?

Reply:
This one looks great. Yes, I recommend doing a different integration. That integration would receive two params in bytes the address of the vault like the best yield plus a number. If the number is 0 then it is the junior tranche if the number is one it's the senior tranche. In the post-action call data for enter I would stake the tokens and in the post-action call data for exit I would unstake (it also claims).

Also take into account that for the convex ones, you would need several operations (curve->convex->idle tranche). See the yearn example to see how you can chain them

Reply:
Sounds good! Shouldn't be an issue - just wanted to pick something simple to start. 

Btw, one last thing I realized - I can't quite tell if there are additional farming rewards for these "best yield" vaults. This makes me think that there are:

<img width="709" alt="Screen Shot 2022-05-28 at 3 59 18 PM" src="https://user-images.githubusercontent.com/1270317/170845302-b793a516-6d03-4bda-ac7e-1c813e40cec1.png">


I'm not sure exactly how this works, as I didn't see it in the interface. But it looks like upon calling `redeemIdleToken`, you actually receive various other tokens? See this tx for an example: <https://etherscan.io/tx/0x98311622aea3211302abc9ffe34559ba58735d36dc0b204d21ec54d37a9039f3>

Looks like I might need to update this PR.

Reply:
Yes, you need to implement this function:

```
    function _getRewardTokens(
        address /* _data */
    ) internal view virtual returns (address[] memory) {
        return new address[](1);
    }
```

Pass the idle token plus the specific governance tokens of every vault

Reply:
Should be all set now 👍

Reply:
> Given that these two integrations are kind of the same but it's more work for you, wdyt about getting a 7.5k bounty in BABL for both?

Missed this part - sounds good!